### ISSUE TITLE
(Sogoban) Unintended sound effects are played infinitely when the mouse hovers in a specific section.

### ISSUE THREAD

**Issue by Jmousy** [ruffle-rs/ruffle#4402]
## Describe the bug
When hovering over the Play button in Sogoban, no sound effects are played in Adobe Flash Player, but in Ruffle, the sound effects used in the menu shown in the next frame play indefinitely.

[Sogoban.zip](https://github.com/ruffle-rs/ruffle/files/6527468/Sogoban.zip)

### 🔊 Original (You can check it after unmute it.)
https://user-images.githubusercontent.com/75052171/119246843-d3b60080-bbbf-11eb-90f5-bec839ac102d.mp4

### 🔊 Ruffle (You can check it after unmute it.)
https://user-images.githubusercontent.com/75052171/119246844-d9abe180-bbbf-11eb-8ba8-cc5ba4940564.mp4

## Expected behavior
When hovering the mouse, the sound effect should be output only once.

## Is the problem with the Ruffle desktop app, extension, or self-hosted version?
Tested on self-hosted.

## What platform are you using?
 - OS: Windows 10 20H2 x64
 - Browser: Chrome Latest
 - Ruffle: 2021-05-23 Nightly

**Comment by Toad06** [comment]
Thanks for the report.

This button (`scripts\DefineButton2 (43)`) uses three different events: press, rollOver and rollOut.

```
on(press){
   _root.gotoAndPlay(4);
}
on(rollOver){
   mvSound = new Sound();
   mvSound.attachSound("move");
   mvSound.start(0,1);
}
on(rollOut){
   rsSound = new Sound();
   rsSound.attachSound("resetsound");
   rsSound.start(0,1);
}
```

There are certainly two distinct issues.

1- The rollOver event is only fired once in Flash Player, but is endlessly triggered in Ruffle with this button. In `frames\frame 1\PlaceObject2 (43) Depth: 171 (playButt)`, changing the depth of the button to 10 (for instance) fixes the issue.

2- Even though the code in the rollOver and rollOut events says to play a sound, Flash Player doesn't play anything. This is really weird, as the other buttons in the menu are using the exact same code in their rollOver and rollOut events, and Flash Player plays a sound this time. I have to admit that I've no clue of what's going on here.

**Comment by Toad06** [comment]
Actually I think that this is the same cause as #1108. The "Play" button is placed in frame 1, then removed in frame 2, then created again in frame 1, etc. Ruffle triggers the `rollOver` event (the one that plays the sound) every time the button is created, which is definitely not what Flash does. :)

However unlike #1108, the button is easy to click because it relies on a `press` event (mouse down), not a `release` event (mouse up).

**Comment by n0samu** [comment]
Fixed by #10862
